modules/widgets/wordpress-post-widget.php

```diff
@@ -79,7 +79,7 @@ function jetpack_display_posts_update_cron_action() {
  * @param string $plugin_file_name The plugin file that was activated.
  */
 function jetpack_conditionally_activate_cron_on_plugin_activation( $plugin_file_name ) {
-	if ( 'jetpack/jetpack.php' === $plugin_file_name ) {
+	if ( plugin_basename(JETPACK__PLUGIN_FILE) === $plugin_file_name ) {
```

Comment:
Actually it does get run, took me a bit digging to figure out how it was working:

When you run the activation action, Jetpack gets activated, but while doing so it gets loaded up - `jetpack.php` is included and all the hooks are added that are there, including the `plugins_loaded` hook, which runs `Jetpack::load_modules` and if Extra Sidebar Widgets is activated, it will include the `wordpress-post-widget.php` file, adding the hook, mentioned below

The thing is that this happens after the hook in `register_activation_hook('jetpack/jetpack.php')` is called, which means that it is of no use. But after digging a bit in the WP core code, it seems that the `activated_plugin` action is called after Jetpack has been initialized after the activation. This means that we get the action where we can handle it in an inner part of the plugin! :)

That's why I [hooked](https://github.com/bisko/jetpack/blob/fb0c25b6830d630cd594189a1d0d1bab25a8b851/modules/widgets/wordpress-post-widget.php#L74) this method to the `activated_plugin` action.